### PR TITLE
RWA-244: Idam pact test

### PR DESCRIPTION
**Pull request by alexbaeza**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-244

### Change description ###

Added IDAM pact test

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/IdamConsumerTest.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    private static final String IDAM_OPENID_USERINFO_URL = "/o/userinfo";
+
+    private static final String PACT_TEST_EMAIL_VALUE = "ia-caseofficer@fake.hmcts.net";
+    private static final String PACT_TEST_PASSWORD_VALUE = "London01";
+    private static final String PACT_TEST_CLIENT_ID_VALUE = "pact";
+    private static final String PACT_TEST_CLIENT_SECRET_VALUE = "pactsecret";
+    private static final String PACT_TEST_SCOPES_VALUE = "openid profile roles";
+    private static final String PACT_TEST_ROLES_VALUE = "caseworker";
+
+    @Pact(provider = "Idam_api", consumer = "wa_task_management_api")
```
Comment (iguannago) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/IdamConsumerTest.java:48]: is this capital letter on purpose: Idam_api ?

**Review by iguannago — DISMISSED** [review]

**Review by lewis150193 — APPROVED** [review]

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/IdamConsumerTest.java (hunk 1 above)
Comment (BenNzewi) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/IdamConsumerTest.java:48]: It appears we are referring to an out of date implementation of case-api. I spoke to Satya and we should use the implementation on the following ia-case-api branch  "ConsumerContractTests".

For example;
In @Pact(provider = "Idam_api", consumer = "wa_task_management_api"), Idam_api should be idamApi_oidc

**Review by satyachundur — COMMENTED** [review]
Currently idam pact verification tests are going to change once this PR is merged https://github.com/hmcts/idam-api/pull/820.  So this old way of implementation might not work with provider names and approach.  Please see this new approach for example https://github.com/hmcts/ia-case-api/tree/ConsumerContractTests/src/contractTest/java/uk/gov/hmcts/reform/iacaseapi/consumer/idam. thanks

File: build.gradle [hunk 2]
```diff
@@ -20,6 +20,7 @@ plugins {
   id 'info.solidsoft.pitest' version '1.5.2'
   id 'uk.gov.hmcts.java' version '0.12.5'
   id 'io.freefair.lombok' version '5.3.0'
+  id 'au.com.dius.pact' version '4.0.4'
```
Comment (BenNzewi) [inline build.gradle:23]: Please reference the  following ia-case-api branch "ConsumerContractTests". It's the most up to date PACT change for case-api.

version should be ->  id 'au.com.dius.pact' version '4.1.0'

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -42,6 +44,7 @@ withPipeline(type, product, component) {
     env.IDAM_API_URL = "https://idam-api.aat.platform.hmcts.net"
     env.CCD_URL = "http://ccd-data-store-api-aat.service.core-compute-aat.internal"
     env.ROLE_ASSIGNMENT_URL = "http://am-role-assignment-service-aat.service.core-compute-aat.internal"
```
Comment (BenNzewi) [inline Jenkinsfile_CNP:46]: i think we can remove env.ROLE_ASSIGNMENT_URL

**Review by lewis150193 — APPROVED** [review]

**Pull request merged**